sslr-core/src/main/java/org/sonar/sslr/channel/CodeBuffer.java

```diff
@@ -59,24 +59,27 @@ protected CodeBuffer(String code, CodeReaderConfiguration configuration) {
    * Note that this constructor will read everything from reader and will close it.
    */
   protected CodeBuffer(Reader initialCodeReader, CodeReaderConfiguration configuration) {
-    Reader reader = null;
 
-    try {
+    /* Make sure the reader passed-in gets closed when done. */
+    try (Reader reader = initialCodeReader) {
```

Comment:
This seems unnecessary:
* it will be closed when final reader closed
* opposite will mean that there is leak anyway
* there is nothing prior to the next `try` except construction of chain